pkg/pushsync/pushsync.go

```diff
@@ -130,6 +143,54 @@ func (ps *PushSync) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream)
 				return fmt.Errorf("chunk store: %w", err)
 			}
 
+			// push the chunk to the closest peers in parallel for replication
+			// any errors here should NOT impact the rest of the handler
+			var skipPeers []swarm.Address
+			for i := 0; i < nPeersToPushsync; i++ {
+				peer, err := ps.closestPeer(chunk.Address(), skipPeers, true)
+				if err != nil {
+					ps.logger.Tracef("closest peer: %w", err)
+					break
+				}
+				skipPeers = append(skipPeers, peer)
+				go func(peer swarm.Address) {
+					receiptPrice := ps.pricer.PeerPrice(peer, chunk.Address())
+					if err := ps.accounting.Reserve(ctx, peer, receiptPrice); err != nil {
+						ps.logger.Tracef("reserve balance for peer %s: %w", peer.String(), err)
```

Comment:
You don't need to log explicitly in every error case. Consider having a variable defined at the beginning of the closure, then a defer call that checks the value of it, logging if necessary:
```go
var err error
defer func() {
if err != nil {
 ps.logger.Tracef("pushsync replication: %v",err)
}
....
err = ps.someCall()
if err != nil { return }
... 

```